src/model/Publication.ts

```diff
@@ -246,45 +246,47 @@ export default class Publication extends R2Publication {
         });
         startPosition = startPosition + positionCount;
       }
-    });
 
-    // Once you have all the positions, you can update all the progressions and total progressions and remaining.
-    positions.map((locator, _index) => {
-      const resource = positions.filter(
-        (el: Locator) => el.href === decodeURI(locator.href)
-      );
-      const positionIndex = Math.ceil(
-        locator.locations.progression * (resource.length - 1)
-      );
-      locator.locations.totalProgression =
-        (locator.locations.position - 1) / positions.length;
-      locator.locations.remainingPositions = Math.abs(
-        positionIndex - (resource.length - 1)
-      );
-      locator.locations.totalRemainingPositions = Math.abs(
-        locator.locations.position - 1 - (positions.length - 1)
-      );
-      return locator;
-    });
+      if (index + 1 === this.readingOrder.length) {
+        // update the link.contentWeight to be a portion of the total and
+        // build up a map of link weights for non fixed layout publications
+        var totalweight = 0;
+        if (this.isReflowable) {
+          this.readingOrder.map(async (link) => {
+            // I don't totally know what this formula is saying haha, maybe we can
+            // add a comment
+            (link as Link).contentWeight =
+              (100 / totalContentLength) * (link as Link).contentLength;
+            totalweight = totalweight + (link as Link).contentWeight;
+          });
+        }
 
-    // update the link.contentWeight to be a portion of the total and
-    // build up a map of link weights for non fixed layout publications
-    // @aferditamuriqi but what is the "weight" object ultimately used for?
-    const weight = {};
-    if (this.isReflowable) {
-      this.readingOrder.forEach((link) => {
-        // I don't totally know what this formula is saying haha, maybe we can
-        // add a comment
-        link.contentWeight = (100 / totalContentLength) * link.contentLength;
-        weight[link.Href] = link.contentWeight;
-      });
-    }
+        // Once you have all the positions, you can update all the progressions and total progressions and remaining.
+        positions.map((locator, _index) => {
+          const resource = positions.filter(
+            (el: Locator) => el.href === decodeURI(locator.href)
+          );
+          const positionIndex = Math.ceil(
+            locator.locations.progression * (resource.length - 1)
+          );
+          locator.locations.totalProgression =
+            (locator.locations.position - 1) / positions.length;
+          locator.locations.remainingPositions = Math.abs(
+            positionIndex - (resource.length - 1)
+          );
+          locator.locations.totalRemainingPositions = Math.abs(
+            locator.locations.position - 1 - (positions.length - 1)
+          );
+          return locator;
+        });
+
+        this.positions = positions;
+      }
+    });
```

Comment:
Same with this, can't this be done after the `map` loop instead of testing that we are on the last item?

Reply:
no it couldn't, it didn't work, the positions and weight where simply empty, i've added commit notes here i beleive, why i added it back into the loop

Reply:
Gotcha, I opened a ticket to do this work later: https://github.com/d-i-t-a/R2D2BC/issues/216